airflow/providers/google/cloud/hooks/datafusion.py

```diff
@@ -0,0 +1,297 @@
+            https://cloud.google.com/data-fusion/docs/reference/rest/v1beta1/projects.locations.instances#Instance
+        :type instance: Dict[str, Any]
+        :param update_mask: Field mask is used to specify the fields that the update will overwrite
+            in an instance resource. The fields specified in the updateMask are relative to the resource,
+            not the full request. A field will be overwritten if it is in the mask. If the user does not
+            provide a mask, all the supported fields (labels and options currently) will be overwritten.
+            A comma-separated list of fully qualified names of fields. Example: "user.displayName,photo".
+            https://developers.google.com/protocol-buffers/docs/reference/google.protobuf?_ga=2.205612571.-968688242.1573564810#google.protobuf.FieldMask
+        :type update_mask: str
+        :param location: The Cloud Data Fusion location in which to handle the request.
+        :type location: str
+        :param project_id: The ID of the Google Cloud Platform project that the instance belongs to.
+        :type project_id: str
+        """
+        if not project_id:
+            raise ValueError("The project_id should be set")
+
+        operation = (
+            self.get_conn()  # pylint: disable=no-member
+            .projects()
+            .locations()
+            .instances()
+            .patch(
+                name=self._name(project_id, location, instance_name),
+                updateMask=update_mask,
+                body=instance,
+            )
+            .execute(num_retries=self.num_retries)
+        )
+        return operation
+
+    def create_pipeline(
+        self,
+        pipeline_name: str,
+        pipeline: Dict[str, Any],
+        instance_url: str,
+        namespace: str = "default",
+    ):
+        """
+        Creates a Cloud Data Fusion pipeline.
+
+        :param pipeline_name: Your pipeline name.
+        :type pipeline_name: str
+        :param pipeline: The pipeline definition. For more information check:
+            https://docs.cdap.io/cdap/current/en/developer-manual/pipelines/developing-pipelines.html#pipeline-configuration-file-format
+        :type pipeline: Dict[str, Any]
+        :param instance_url: Endpoint on which the REST APIs is accessible for the instance.
+        :type instance_url: str
+        :param namespace: f your pipeline belongs to a Basic edition instance, the namespace ID
+            is always default. If your pipeline belongs to an Enterprise edition instance, you
+            can create a namespace.
+        :type namespace: str
+        """
+        url = os.path.join(
+            instance_url, "v3", "namespaces", namespace, "apps", pipeline_name
+        )
+        response = self._cdap_request(url=url, method="PUT", body=pipeline)
+        if response.status != 200:
+            raise AirflowException(
+                f"Creating a pipeline failed with code {response.status}"
+            )
```

Comment:
@jayfjin please take a look. The `instance_url` is the `apiEndpoint` of and instance.